### PR TITLE
CalendarDateGroupingExpr

### PR DESCRIPTION
**Pull request by microArtur**
https://nanopay.atlassian.net/browse/NP-6296

### CODE REVIEW

File: src/foam/mlang/expr/CalendarDateGroupingExpr.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            return a.getLow() - b.getHigh();
+          }
+        });
+
+        CreatedAware createdAwareObj = (CreatedAware) obj;
+
+        Date today = new Date();
+        Date objDate = new Date(createdAwareObj.getCreated().getTime());
+
+        Date d1 = new Date(objDate.getYear() + 1900, objDate.getMonth(), objDate.getDate());
```
Comment (jlhughes) [inline src/foam/mlang/expr/CalendarDateGroupingExpr.js]: What is going on here?  getYear is only give two digits?

**Comment by jlhughes** [comment]
@microArtur how and where is this being used? Ignore, I read the ticket.

File: src/menus.jrl [hunk 2]
```diff
@@ -272,7 +272,7 @@ p({
           "class":"foam.nanos.notification.NotificationRowView"
         },
         "groupExpr":{
-          "class":"foam.mlang.expr.DateTimeGroupingExpr",
+          "class":"foam.mlang.expr.CalendarDateGroupingExpr",
```
Comment (Nauna) [inline src/menus.jrl]: So now we have CalendarDateGroupingExpr, but there is also:
DateTimeGroupingExpr
DateGroupingExpr

it does seem - at least by the name that - that one of these is repeated..

Comment (Nauna) [reply src/menus.jrl]: Hmmm - gotta even wonder - was this even still an issue - https://nanopay.atlassian.net/browse/NP-6296 was created in Jan 2022, and Tala did the above update to MyNotificationDAO in March (https://github.com/kgrgreer/foam3/pull/1495).

Comment (jlhughes) [reply src/menus.jrl]: yes, it's messy.  
What I would prefer is 
1. replacing DateTyps with TimeUnit
2. add the array property to DateTimeGrouping 
3. make the default implementation work on human days when the unit DAY or greater.

**Comment by kgrgreer** [comment]
Fixed in:

commit 68a5aa2c3209c831c43feb1cf8aef7642fb8a092
Author: Kevin Greer <kgrgreer@gmail.com>
Date:   Wed Jul 20 14:14:47 2022 -0400

    Fixes NP-6296, Notifications not grouped correctly.

**Pull request closed without merging**